packages/dev/core/src/XR/webXRExperienceHelper.ts

```diff
@@ -222,15 +237,37 @@ export class WebXRExperienceHelper implements IDisposable {
      * display the first rig camera's view on the desktop canvas.
      * Please note that this will degrade performance, as it requires another camera render.
      * It is also not recommended to enable this in devices like the quest, as it brings no benefit there.
+     * @param options giving WebXRSpectatorModeOption for specutator camera to setup when the spectator mode is enabled.
      */
-    public enableSpectatorMode(): void {
-        if (!this._spectatorMode) {
-            const updateSpectatorCamera = () => {
-                if (this._spectatorCamera) {
-                    this._spectatorCamera.position.copyFrom(this.camera.rigCameras[0].globalPosition);
-                    this._spectatorCamera.rotationQuaternion.copyFrom(this.camera.rigCameras[0].absoluteRotation);
+    public enableSpectatorMode(options?: WebXRSpectatorModeOption): void {
+        this._spectatorMode = true;
+        this._switchSpectatorMode(options);

```

Comment:
please only call the switch function when `this._spectatorMode` has changed (in this case when it is false).
```suggestion
        if(!this._spectatorMode) {
            this._spectatorMode = true;
            this._switchSpectatorMode(options);
        }
```

Reply:
Resolved by ad0fbd46bd3c72d9e1aeb0db9f610a408df2e244